lib/component.js

```diff
@@ -383,9 +383,13 @@ class Component extends WebComponent {
   // a full state update to one component while sending only "shared" state
   // updates to the app root.
   updateSelfAndChildren(stateUpdate, options={}) {
+    if (!this.initialized) {
+      return;
+    }
+
     const {store, cascade} = options;
```

Comment:
Platform apps have their own version of mixpanel-common which has its own version of panel, so things blow up when options is an empty object.

Should we atleast default the store and cascade if they are not present?

```
const {store = `state`, cascade = true} = options
```

```
const stateStore = this[store];
Object.assign(stateStore, stateUpdate);
if (store !== `state` || this.shouldUpdate(stateStore)) {
```

wdyt ?

Reply:
I kind of think that this lib code shouldn't be defensive against a poor integration (multiple incompatible versions of the lib living in the same component hierarchy). The right thing is to be sure that all linked components are running the same code. In practice with the `mixpanel-common` thing, the simplest solution would be to export `panel` from within `mixpanel-common` so that an app can inherit from its classes or make its own components and be sure that there's only a single version of the Panel code. This is the same concept as how Panel exports the vdom `h()` function so that you don't have to worry about the installation/versioning of the virtual dom lib.

Reply:
Works for me